src/lib/openapi/spec/archived-feature-schema.ts

```diff
@@ -0,0 +1,103 @@
+import type { FromSchema } from 'json-schema-to-ts';
+
+export const archivedFeatureSchema = {
+    $id: '#/components/schemas/archivedFeatureSchema',
+    type: 'object',
+    additionalProperties: true,
+    required: ['name', 'project'],
```

Comment:
Can more of the properties be required? Should they? I'm not sure where we stand here in general, though. But I'm assuming that most (all?) properties on here will always be present, which is why description is `nullable`, for instance?

Marking properties as required also makes them come through as not potentially undefined in the front end API, which is useful for our sake.

Reply:
I made all fields required but then TS checks failed for the types used in the services
```
export interface FeatureToggleDTO {
    name: string;
    description?: string;
    type?: string;
    stale?: boolean;
    archived?: boolean;
    archivedAt?: Date;
    createdAt?: Date;
    impressionData?: boolean;
    variants?: IVariant[];
    tags?: ITag[];
    createdByUserId?: number;
    createdBy?: {
        id: number;
        name: string;
        imageUrl: string;
    };
}
```
By making those optional I can merge my changes (splitting one schema into two schemas to remove fields we never return) and over time we can improve other types. Making it more strict over time is easy since it's not a breaking change.